schema/rules.go

```diff
@@ -43,12 +44,43 @@ type SearchIndexValidator interface {
 	ValidateIndex(existing *SearchIndex, current *SearchFactory) error
 }
 
-type IndexSchemaValidator struct{}
+type PrimaryIndexSchemaValidator struct{}
 
-func (v *IndexSchemaValidator) Validate(existing *DefaultCollection, current *Factory) error {
+func (v *PrimaryIndexSchemaValidator) Validate(existing *DefaultCollection, current *Factory) error {
 	return existing.Indexes.PrimaryKey.IsCompatible(current.Indexes.PrimaryKey)
 }
 
+type SecondaryIndexSchemaValidator struct{}
```

Comment:
This is good and is for backward compatibility. Also, add a check in fieldBuilder `Build` method which is done when we receive the schema and build the fields.